src/objectExplorer/objectExplorerUtils.ts

```diff
@@ -32,6 +32,11 @@ export class ObjectExplorerUtils {
 
     public static getNodeUriFromProfile(profile: IConnectionProfile): string {
         let uri: string;
+        if (profile.connectionString) {
+            let fields = profile.connectionString.split(';').filter(s => !s.toLowerCase().includes('password'));
```

Comment:
Edge case - but what if the password contains a semi-colon? Reading around it seems like {} is used to escape it (and then double up }'s to escape those). 

Might be worth looking into a library to do this for you. I'm not a huge fan of adding dependencies for one small edge case but possibly you could at least see what they do and handle it the same way.

Reply:
Will add additional logic for edge case in a separate PR